src/main/kotlin/org/rust/ide/sdk/flavors/RsSdkFlavor.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.sdk.flavors
+
+import com.intellij.openapi.extensions.ExtensionPointName
+import com.intellij.openapi.util.SystemInfo
+import com.intellij.util.io.isDirectory
+import org.rust.cargo.toolchain.tools.Cargo
+import org.rust.cargo.toolchain.tools.Rustc
+import org.rust.stdext.isExecutable
+import java.nio.file.Path
+
+fun RsSdkFlavor.suggestHomePaths(): List<Path> =
+    getHomePathCandidates().distinct().filter { isValidSdkPath(it) }
+
+interface RsSdkFlavor {
+
+    /**
+     * Flavor is added to result in [getApplicableFlavors] if this method returns true.
+     * @return whether this flavor is applicable.
+     */
+    fun isApplicable(): Boolean = true
+
+    fun getHomePathCandidates(): List<Path>
```

Comment:
Looks like this method is not a part of public API that we want to provide since it should be use only in combination with `isValidSdkPath`
Maybe it's better to convert `RsSdkFlavor` to abstract class and made this method protected?